Source/Core/Core/PowerPC/Expression.cpp

```diff
@@ -225,7 +228,13 @@ double Expression::Evaluate() const
 {
   SynchronizeBindings(SynchronizeDirection::From);
 
-  double result = expr_eval(m_expr.get());
+  double result;
+  {
+    Core::CPUThreadGuard guard;
+    m_expr->param.func.context = &guard;
```

Comment:
This only affects the top level of the expression tree (and only if it happens to be a function call), so this mostly isn't working as intended. The concept of a context in the expr library doesn't map nicely to what you want to do here. If this were something performance critical, I think the right fix would be to add a context parameter to expr_eval(). However, given that constructing guards is so cheap after the initial lock, rather than trying so hard to pass this one around it should be fine to just construct additional guards in the functions that need them.

Reply:
Thanks for catching that! I wasn't entirely sure if I was correctly understanding how the context worked.

Addressed in PR #11602.